conformance/lang/expressions/let-expr/let_expr.balt

```diff
@@ -0,0 +1,593 @@
+Test-Case: output
+Description: Test nil-literal and nil-type in let-var-decl.
+Labels: let-expr, nil-literal, nil-type, function-call-expr
+
+function init() {
+    io:println(let () n = () in n.toBalString()); // @output ()
+    io:println(let null n = () in n.toBalString()); // @output ()
+}
+
+Test-Case: output
+Description: Test nil-type, boolean and string in let-var-decl.
+Labels: let-expr, nil-literal, nil-type, boolean, boolean-literal, string, function-call-expr
+
+function init() {
+    io:println(let null n = null in n.toBalString()); // @output ()
+    io:println(let boolean b = true in b); // @output true
+    io:println(let string s = "\u{0633}" in s); // @output س
+    io:println(let string s = "ABC \u{0633} CDE \u{0644} DEF \u{0644} XYZ" in s); // @output ABC س CDE ل DEF ل XYZ
```

Comment:
Still combines multiple things. NumericEscapes aren't a basic concept IMO, and can/should be a separate test.

Reply:
Also need the `StringNumericEscape` label.

Reply:
Changed